src/comestible_inv.cpp

```diff
@@ -0,0 +1,780 @@
+                                                       ctxt.get_desc( "SHOW_HELP" ) );
+                mvwprintz( head, point( w_width - ( minimap_width + 2 ) - utf8_width( msg ) - 1, 0 ),
+                           c_white, msg );
+            }
+            if( g->u.has_watch() ) {
+                const std::string time = to_string_time_of_day( calendar::turn );
+                mvwprintz( head, point( 2, 0 ), c_white, time );
+            }
+            wrefresh( head );
+
+            redraw_minimap();
+        }
+        refresh( false, false );
+
+        process_actions( ctxt );
+    }
+}
+
+void comestible_inventory::redraw_minimap()
+{
+    werase( mm_border );
+    // redraw border around minimap
+    draw_border( mm_border );
+    // minor addition to border for AIM_ALL, sorta hacky
+    if( pane->get_area()->info.type == comestible_inv_area_info::AREA_TYPE_MULTI ) {
+        //point(1,0) - point_east so travis is happy
+        mvwprintz( mm_border, point_east, c_light_gray, utf8_truncate( _( "Mul" ), minimap_width ) );
+    }
+    wrefresh( mm_border );
+
+    werase( minimap );
+    // get the center of the window
+    tripoint pc = { getmaxx( minimap ) / 2, getmaxy( minimap ) / 2, 0 };
+    // draw the 3x3 tiles centered around player
+    g->m.draw( minimap, g->u.pos() );
+
+    if( pane->get_area()->info.type == comestible_inv_area_info::AREA_TYPE_MULTI ) {
+        return;
+    }
+
+    char sym = minimap_get_sym();
+    if( sym != '\0' ) {
+        auto sq = pane->get_area();
+        auto pt = pc + sq->offset;
+        // invert the color if pointing to the player's position
+        auto cl = sq->info.type == comestible_inv_area_info::AREA_TYPE_PLAYER ?
+                  invert_color( c_light_cyan ) : c_light_cyan.blink();
+        mvwputch( minimap, pt.xy(), cl, sym );
+    }
+
+    // Invert player's tile color if exactly one pane points to player's tile
+    bool player_selected = pane->get_area()->info.type == comestible_inv_area_info::AREA_TYPE_PLAYER;
+    g->u.draw( minimap, g->u.pos(), player_selected );
+    wrefresh( minimap );
+}
+
+char comestible_inventory::minimap_get_sym() const
+{
+    if( pane->get_area()->info.type == comestible_inv_area_info::AREA_TYPE_PLAYER ) {
+        return '^';
+    } else if( pane->is_in_vehicle() ) {
+        return 'V';
+    } else {
+        return ' ';
+    }
+}
+
+void comestible_inventory::do_return_entry()
+{
+    // only save pane settings
+    g->u.assign_activity( activity_id( "ACT_COMESTIBLE_INVENTORY" ) );
+    g->u.activity.auto_resume = true;
+    uistate.comestible_save.exit_code = exit_re_entry;
+    save_settings();
+}
+
+void comestible_inventory::refresh( bool needs_recalc, bool needs_redraw )
+{
+    recalc = needs_recalc;
+    pane->needs_recalc = needs_recalc;
+
+    redraw = needs_redraw;
+    pane->needs_redraw = needs_redraw;
+}
+
+void comestible_inventory_food::init()
+{
+    pane = new comestible_inventory_pane_food();
```

Comment:
You could forward a reference to `squares` to the new pane here, instead of forwarding it via `init`. It could be stored as a reference inside the `comestible_inventory_pane`.

Reply:
If I put it here, then I will need to do it in every child class, so the common initialization is duplicated. Seems bad.

Reply:
> so the common initialization is duplicated. Seems bad.

The common initialization is already duplicated in the `init` functions. Forwarding it via the constructor allows you to make it a reference in the `pane`, which means no pointer and no danger of uninitialized pointers.